pxr/imaging/hdSt/materialNetwork.cpp

```diff
@@ -1100,57 +1100,85 @@ HdStMaterialNetwork::ProcessMaterialNetwork(
 {
     HD_TRACE_FUNCTION();
 
-    _fragmentSource.clear();
-    _displacementSource.clear();
-    _materialMetadata.clear();
-    _materialParams.clear();
-    _textureDescriptors.clear();
-    _materialTag = HdStMaterialTagTokens->defaultMaterialTag;
+    bool isVolume = false;
+
+    auto filterTask = std::make_shared<HdSt_MaterialFilterTask>();
+    filterTask->hdNetwork =
+        HdConvertToHdMaterialNetwork2(hdNetworkMap, &isVolume);
 
     // The fragment source comes from the 'surface' network or the
     // 'volume' network.
-    bool isVolume = false;
     HdMaterialNetwork2 surfaceNetwork =
         HdConvertToHdMaterialNetwork2(hdNetworkMap, &isVolume);
     const TfToken &terminalName = (isVolume) ? HdMaterialTerminalTokens->volume 
                                             : HdMaterialTerminalTokens->surface;
 
-    SdfPath surfTerminalPath;
-    if (HdMaterialNode2 const* surfTerminal = 
-            _GetTerminalNode(surfaceNetwork, terminalName, &surfTerminalPath)) {
+    filterTask->terminalNode = 
```

Comment:
Here & material.*, instead of bringing the "filterTask" API into a bunch of files in Storm, you could modularize things better by either:
(1) having the scene index produce a "glsl" material network that can be read by the non-materialx code; or
(2) having the scene index produce a digested material network as code snippets for each stage & params, and pass them through as datasources.  If material.cpp finds that, it uses it, and otherwise it falls back to network processing.

This splits the code up a little better, which is nice since the MaterialX code is an external dependency with an evolving API and hidden behind a build flag, so having a big API interface between the two makes me a bit worried.